crates/store/re_data_loader/src/loader_lerobot.rs

```diff
@@ -0,0 +1,270 @@
+use anyhow::{anyhow, Context};
+use arrow::array::{FixedSizeListArray, Int64Array, RecordBatch};
+use arrow::compute::cast;
+use arrow::datatypes::{DataType, Field};
+use itertools::Either;
+use re_arrow_util::ArrowArrayDowncastRef;
+use re_chunk::external::nohash_hasher::IntMap;
+use re_chunk::{ArrowArray, Chunk, ChunkId, RowId, TimeColumn, TimePoint, Timeline};
+
+use re_log_types::StoreId;
+use re_types::archetypes::{AssetVideo, VideoFrameReference};
+use re_types::components::{Scalar, VideoTimestamp};
+use re_types::{Archetype, Component, ComponentBatch};
+
+use crate::le_robot::{DType, LeRobotDataset};
+use crate::{DataLoader, DataLoaderError, LoadedData};
+
+pub struct LeRobotDatasetLoader;
+
+impl DataLoader for LeRobotDatasetLoader {
+    fn name(&self) -> String {
+        "LeRobotDatasetLoader".into()
+    }
+
+    fn load_from_path(
+        &self,
+        _settings: &crate::DataLoaderSettings,
+        filepath: std::path::PathBuf,
+        tx: std::sync::mpsc::Sender<LoadedData>,
+    ) -> Result<(), DataLoaderError> {
+        if !crate::le_robot::is_le_robot_dataset(&filepath) {
+            return Err(DataLoaderError::Incompatible(filepath));
+        }
+
+        let dataset = LeRobotDataset::load_from_directory(&filepath)
+            .with_context(|| "Loading LeRobot dataset failed")?;
+
+        re_log::info!(
+            "Loading LeRobot dataset from `{:?}`, with {} episode(s)",
+            dataset.path,
+            dataset.metadata.episodes.len()
+        );
+
+        for episode in &dataset.metadata.episodes {
+            let episode_idx = episode.episode_index;
+            let store_id = StoreId::from_string(
+                re_log_types::StoreKind::Recording,
+                format!("episode_{episode_idx}"),
+            );
+
+            let data = dataset
+                .read_episode_data(episode_idx)
+                .with_context(|| format!("Reading data for episode {episode_idx} failed!"))?;
+
+            let frame_indices = data
+                .column_by_name("frame_index")
+                .ok_or_else(|| anyhow!("failed to get frame index"))?
+                .clone();
+
+            let timeline = re_log_types::Timeline::new_sequence("frame_index");
+            let times: &arrow::buffer::ScalarBuffer<i64> = frame_indices
+                .downcast_array_ref::<Int64Array>()
+                .ok_or_else(|| anyhow!("frame indices are of an unexpected type"))?
+                .values();
+
+            let time_column = re_chunk::TimeColumn::new(Some(true), timeline, times.clone());
```

Comment:
Are we _sure_ it is sorted? How can we be sure?

Reply:
We're not, and there's no harm in sorting these